File: src/device_profiles.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True, slots=True)
class DeviceProfile:
    key: str
    label: str
    width: int
    height: int


DEVICE_PROFILES = [
    DeviceProfile("iphone_15_pro", "iPhone 15 Pro", 1179, 2556),
    DeviceProfile("iphone_15_pro_max", "iPhone 15 Pro Max", 1290, 2796),
    DeviceProfile("iphone_14", "iPhone 14 / 13 / 12", 1170, 2532),
    DeviceProfile("iphone_14_plus", "iPhone 14 Plus / 13 Pro Max / 12 Pro Max", 1284, 2778),
    DeviceProfile("iphone_se", "iPhone SE / 8 / 7 / 6", 750, 1334),
    DeviceProfile("custom", "Custom", 1179, 2556),
]

DEVICE_PROFILE_MAP = {profile.key: profile for profile in DEVICE_PROFILES}


def get_device_profile(key: str) -> DeviceProfile:
    return DEVICE_PROFILE_MAP.get(key, DEVICE_PROFILE_MAP["iphone_15_pro"])

File: web/src/deviceProfiles.js
export const deviceProfiles = [
  { key: 'iphone_17', label: 'iPhone 17', width: 1206, height: 2622 },
  { key: 'iphone_17_air', label: 'iPhone 17 Air', width: 1260, height: 2736 },
  { key: 'iphone_17_pro', label: 'iPhone 17 Pro', width: 1206, height: 2622 },
  { key: 'iphone_17_pro_max', label: 'iPhone 17 Pro Max', width: 1320, height: 2868 },
  { key: 'iphone_17e', label: 'iPhone 17e', width: 1170, height: 2532 },
  { key: 'iphone_16', label: 'iPhone 16', width: 1179, height: 2556 },
  { key: 'iphone_16_plus', label: 'iPhone 16 Plus', width: 1290, height: 2796 },
  { key: 'iphone_16_pro', label: 'iPhone 16 Pro', width: 1206, height: 2622 },
  { key: 'iphone_16_pro_max', label: 'iPhone 16 Pro Max', width: 1320, height: 2868 },
  { key: 'custom', label: 'Custom', width: 1206, height: 2622 },
]

export function getDeviceProfile(key) {
  return deviceProfiles.find((profile) => profile.key === key) ?? deviceProfiles[2]
}
